scraper.py

```diff
@@ -22,30 +27,159 @@
+    return rdata
+
+
+def is_inspection_row(element):
+    is_row = element.name == 'tr'
+    if not is_row:
+        return False
+    cells = element.find_all('td', recursive=False)
+    has_four = len(cells) == 4
+    this_text = clean_data(cells[0]).lower()
+    contains_word = 'inspection' in this_text
+    does_not_start = not this_text.startswith('inspection')
+    return is_row and has_four and contains_word and does_not_start
+
+
+def extract_score_data(element):
+    inspection_rows = element.find_all(is_inspection_row)
+    samples = len(inspection_rows)
+    total = high_score = average = 0
+    for row in inspection_rows:
+        string_value = clean_data(row.find_all('td')[2])
+        try:
+            int_value = int(string_value)
+        except (ValueError, TypeError):
+            samples -= 1
+        else:
+            total += int_value
+            high_score = int_value if int_value > high_score else high_score
+    if samples:
+        average = total/float(samples)
+    data = {
+        u'Average Score': average,
+        u'High Score': high_score,
+        u'Total Inspections': samples
+    }
+    return data
+
+
+def generate_results(test=False, count=10):
+    kwargs = {
+        'Inspection_Start': '2/1/2013',
+        'Inspection_End': '2/1/2015',
+        'Zip_Code': '98109'
+    }
+    if test:
+        source, encoding = load_inspection_page()
+    else:
+        source, encoding = get_inspection_page(**kwargs)
+    doc = parse_source(source, encoding)
+    listings = extract_data_listings(doc)
+    for listing in listings[:count]:
+        metadata = extract_restaurant_metadata(listing)
+        score_data = extract_score_data(listing)
+        metadata.update(score_data)
+        yield metadata
+
+
+def get_geojson(result):
+    """Get geocoding data from google using address of the restaurant."""
+    try:
+        address = ' '.join(result['Address'])
+    except KeyError:
+        return None
+    geocoded = geocoder.google(address)
+    geojson = geocoded.geojson
+    keys = ('Business Name',
+            'Average Score',
+            'Total Inspections',
+            'High Score',
+            'Address')
+    inspection_data = {k: result.get(k, '') for k in keys}
+    for key, value in inspection_data.items():
+        if isinstance(value, list):
+            inspection_data[key] = ' '.join(value)
+    new_address = geojson['properties'].get('address')
+    if new_address:
+        inspection_data['Address'] = new_address
+    geojson['properties'] = inspection_data
+    return geojson
 
 
-# with io.open('kc_health_data.html') as f:
-#                html_doc = f.read()
-#
-# soup = BeautifulSoup(html_doc, 'html.parser')
-#
-# print(soup)
+if __name__ == '__main__':
+    test = len(sys.argv) > 1 and sys.argv[1] == 'test'
+    features = [get_geojson(r) for r in generate_results(test)]
+    for feature in features:
+        pprint(feature)
```

Comment:
It would be better if you put this on a command-line switch, so you could print it if you passed '-v' or something like that.